src/PolykeyClient.ts

```diff
@@ -122,24 +122,23 @@ class PolykeyClient {
   } = {}): Promise<void> {
     this.logger.info(`Starting ${this.constructor.name}`);
 
-    const statusPath = path.join(this.nodePath, 'status');
+    const statusPath = path.join(this.nodePath, 'status.json');
     const status = new Status({
       statusPath,
       fs: this.fs,
-      logger: this.logger.getChild('Lockfile'),
+      logger: this.logger.getChild('Status'),
     });
 
     // Wait if the agent is starting.
     const statusConfig = await status.waitFor('LIVE', 20000);
-
     // Attempt to read token from fs and start session.
     await this.session.start();
 
     // Create a new GRPCClientClient
     this._grpcClient = await GRPCClientClient.createGRPCClientClient({
       nodeId: statusConfig.data.nodeId as NodeId,
-      host: host ?? statusConfig.data.host ?? 'localhost',
-      port: port ?? statusConfig.data.port ?? 0,
+      host: host ?? statusConfig.data.clientHost ?? 'localhost',
```

Comment:
These defaults should be the same as that is being used by `PolykeyAgent`.

You cannot default these 2 parameters. These must be passed in from the outside.

Reply:
The `Status` is read only when we don't have the parameters directly available.

Reply:
If neither parameters are passed in, and cannot be read from the status, it's not possible to start.

Reply:
PolykeyClient should not be using `Status`. Instead `Status` should be used by the bin CLI commands and the parameters should be read from there and then  passed to the `PolykeyClient` directly.

Therefore:

1. `PolykeyClient` should not be using `Status`
2. `PolykeyClient` should take in `host` and `port` directly
3. `PolykeyClient` should not be using `nodePath` as a parameter at all

Reply:
I've resolved this, the `PolykeyClient` now has this creation signature:

```ts
  static async createPolykeyClient({
    nodeId,
    host,
    port,
    clientPath,
    session,
    grpcClient,
    timeout,
    fs = require('fs'),
    logger = new Logger(this.name),
    fresh = false,
  }: {
    nodeId: NodeId;
    host: Host;
    port: Port;
    clientPath: string;
    timeout?: number;
    session?: Session;
    grpcClient?: GRPCClientClient;
    fs?: FileSystem;
    logger?: Logger;
    fresh?: boolean;
  }): Promise<PolykeyClient>
```

This means all the `bin` commands and tests using `PolykeyClient` needs to be cleaned up appropriately. They should be acquiring the `Status` directly and reading that data if they want to pass it to `PolykeyClient`.

Reply:
They client shouldn't be embedding status. That's too much complexity in the `PolykeyClient`. This simplifies things so that `PolykeyClient` can be used by other areas that aren't dealing with `Status`. Furthermore the `nodePath` cannot be guaranteed to exist on client users because client and agent can be on different machines.

Reply:
Right now the bin CLI commands utilise the `Commander` options to resolve the node ID, client host, and client port:

```ts
    this.addOption(binOptions.nodeId);
    this.addOption(binOptions.clientHost);
    this.addOption(binOptions.clientPort);
```

This means that we're currently defaulting to what's specified in `options.ts`:

```ts
const nodeId = new commander.Option('-ni', '--node-id <id>').env('PK_NODE_ID');

const clientHost = new commander.Option(
  '-ch, --client-host <address>',
  'Client Host Address',
)
  .env('PK_CLIENT_HOST')
  .default('127.0.0.1');

const clientPort = new commander.Option(
  '-cp, --client-port <port>',
  'Client Port',
)
  .argParser(parsers.parseNumber)
  .env('PK_CLIENT_PORT')
  .default(0);
```

Based on our discussion this morning, it sounds like instead, our "default" value should be what's specified in the `Status` file. If we can't acquire a live status file within some timeout, then we should instead throw an exception (indicating that we can't produce the required client parameters). This would mean that we remove these `.default` settings here.

That is, the precedence should be: CLI argument > environment variable > status file

Reply:
@tegefaulkes and I have added a `resolveClientOptions` utility function to provide support for this retrieval from the status file (if any of node ID, client host, or client port, are undefined).

Reply:
I'll resolve this by creating a `src/bin/utils/processors.ts` that process options, and adding defaults to `src/config.ts`.